packages/microservices/server/server-kafka.ts

```diff
@@ -163,9 +163,10 @@ export class ServerKafka extends Server implements CustomTransportStrategy {
       payload.partition,
       payload.topic,
     ]);
+    const handler = this.getHandlerByPattern(packet.pattern);
     // if the correlation id or reply topic is not set
     // then this is an event (events could still have correlation id)
-    if (!correlationId || !replyTopic) {
+    if ((handler && handler.isEventHandler) || !correlationId || !replyTopic) {
```

Comment:
can't we use [optional chaining](https://www.proposals.es/proposals/Optional%20Chaining)

```suggestion
    if (handler?.isEventHandler || !correlationId || !replyTopic) {
```

```
a?.b  // undefined if `a` is null/undefined, `a.b` otherwise
```

Reply:
thank you :)